source/opt/inst_bindless_check_pass.cpp

```diff
@@ -48,18 +47,29 @@ uint32_t InstBindlessCheckPass::GenDebugReadLength(
 uint32_t InstBindlessCheckPass::GenDebugReadInit(uint32_t var_id,
                                                  uint32_t desc_idx_id,
                                                  InstructionBuilder* builder) {
-  uint32_t desc_set_base_id =
-      builder->GetUintConstantId(kDebugInputBindlessInitOffset);
-  uint32_t desc_set_idx_id = builder->GetUintConstantId(var2desc_set_[var_id]);
   uint32_t binding_idx_id = builder->GetUintConstantId(var2binding_[var_id]);
   uint32_t u_desc_idx_id = GenUintCastCode(desc_idx_id, builder);
-  return GenDebugDirectRead(
-      {desc_set_base_id, desc_set_idx_id, binding_idx_id, u_desc_idx_id},
-      builder);
+  // If desc index checking is not enabled, we know the offset of initialization
+  // entries is 1, so we can avoid loading this value and just add 1 to the
+  // descriptor set.
+  if (!desc_idx_enabled_) {
+    uint32_t desc_set_idx_id =
+        builder->GetUintConstantId(var2desc_set_[var_id] + 1);
+    return GenDebugDirectRead({desc_set_idx_id, binding_idx_id, u_desc_idx_id},
+                              builder);
+  } else {
+    uint32_t desc_set_base_id =
+        builder->GetUintConstantId(kDebugInputBindlessInitOffset);
+    uint32_t desc_set_idx_id =
+        builder->GetUintConstantId(var2desc_set_[var_id]);
+    return GenDebugDirectRead(
+        {desc_set_base_id, desc_set_idx_id, binding_idx_id, u_desc_idx_id},
+        builder);
+  }
 }
 
 uint32_t InstBindlessCheckPass::CloneOriginalReference(
-    ref_analysis* ref, InstructionBuilder* builder) {
+    InstBindlessCheckPass::ref_analysis* ref, InstructionBuilder* builder) {
```

Comment:
The C++ based editor would occasionally get in a mood where it would not recognize ref_analysis here (and in similar places). Adding the class name made it happy. The problem is intermittent. At the moment it is happy, so I will change these back. It feels like there could be a true solution for this.